Code/Framework/AzCore/AzCore/Task/TaskExecutor.cpp

```diff
@@ -57,6 +67,19 @@ namespace AZ
             // TODO: Check for dependency cycles
         }
 
+        CompiledTaskGraph::~CompiledTaskGraph()
+        {
+            // record deallocation breadcrumbs
+            AZStd::scoped_lock<AZStd::mutex> lock(s_deallocationMutex);
+
+            uint32_t deallocationSlot = s_nextDeallocationSlot;
```

Comment:
Looks good, minor nit but I'd probably do something like `DeallocationData& slot = s_recentDeallocations[deallocationSlot++ % NumTrackedRecentDeallocations];` or something to avoid an empty 0 slot on the first frame (since there may be cases where this error happens shortly after launch maybe?)

Reply:
This uses slot 0 on the first entry, it could be shortened to a single line but that doesn't seem like a huge win.

Reply:
Thanks I misread it